bazel/sh_test_wrapper.sh

```diff
@@ -3,4 +3,6 @@
 # Where the runfiles are for tests.
 export TEST_RUNDIR="${TEST_SRCDIR}/${TEST_WORKSPACE}"
 
+cd $(dirname "$0")
```

Comment:
So, a few of things:

1. This might break the Google import, as this is making an assumption about file layout relative to where the source file is. Does something like `cd "${TEST_RUNDIR}"` work instead?

2. There should be a cleaner way to disable the test than totally removing its sources. I still think tags is one way to manage this, if there's a way in `envoy_sh_test` to figure out if we're on OS X or not. https://github.com/lyft/envoy/blob/master/bazel/cc_configure.bzl#L30 does this basically, but does so in a repository rule context. So it's possible in Bazel..

3. Nit: `cd "$(dirname "$0")"